### PR TITLE
Added solutions for chapters 1, 2, and 3

### PR DESCRIPTION
**Pull request by matthewcarroll**
Hello,

I implemented solutions for chapter 2 in Swift 3 using Xcode beta 6. I did them in a playground since playgrounds are nice as an interactive environment to work problems such as these. 

If this is Gayle, I'm really enjoying your book. I appreciate the scope of the topics covered and the treatment of the problems. 

Please let me know what you think.

Thanks!

Matthew

### CODE REVIEW

**Comment by matthewcarroll** [comment]
Hello @asaph,

I'm not sure if @ndavon is online. There have been several pull requests open for a couple of months for this repo. Is there anyone else who could take a look?

Thanks,
Matthew

**Comment by asaph** [comment]
@matthewcarroll Sorry for the delay. If @ndavon isn't available soon, I'll review your Chapter 2 solutions. Disclaimer: I don't have much experience in Swift, but [I've solved all of the Chapter 2 problems in another language](https://github.com/careercup/CtCI-6th-Edition-php/tree/master/src/chapter02) and I use a Mac.

**Comment by matthewcarroll** [comment]
@asaph No problem. Thank you for taking a look!

**Comment by matthewcarroll** [comment]
Merged in 970adac

File: Ch 01. Arrays and Strings/Chapter1.playground/Pages/1.1 Is Unique.xcplaygroundpage/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*:
+ 1.1 Determine if a string has all unique characters. What if you can't use additional data structures?
+ */
+extension String {
+    
+    func hasUniqueCharacters() -> Bool {
```
Comment (tikipatel) [inline Ch 01. Arrays and Strings/Chapter1.playground/Pages/1.1 Is Unique.xcplaygroundpage/Contents.swift]: As a curiosity, what would the performance difference be between your solution and the following:

```func hasUniqueCharacters() -> Bool {
        return characters.count == Set(characters).count
```}

Comment (matthewcarroll) [reply Ch 01. Arrays and Strings/Chapter1.playground/Pages/1.1 Is Unique.xcplaygroundpage/Contents.swift]: The running time of that implementation would always be O(n), even in cases where the set contained duplicate characters. In the implementation above, the loop stops at the first duplicate character.

**Pull request merged**